crypto/cpu-arm-ios.c

```diff
@@ -0,0 +1,27 @@
+// Copyright 2017 Mathieu Poumeyrol
+//
+// Permission to use, copy, modify, and/or distribute this software for any
+// purpose with or without fee is hereby granted, provided that the above
+// copyright notice and this permission notice appear in all copies.
+//
+// THE SOFTWARE IS PROVIDED "AS IS" AND AND THE AUTHORS DISCLAIM ALL WARRANTIES
+// WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+// MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY
+// SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+// WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION
+// OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
+// CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+#include <openssl/cpu.h>
+
+#if defined(OPENSSL_ARM) && !defined(OPENSSL_STATIC_ARMCAP)
+
+#include <openssl/arm_arch.h>
+
+extern uint32_t GFp_armcap_P;
+
+void GFp_cpuid_setup(void) {
+    GFp_armcap_P |= ARMV7_NEON;
```

Comment:
We shouldn't need to implement `GFp_cpuid_setup` for NEON on non-Android targets that have a C compiler with NEON enabled. Please rebase your assembly language changes on top of https://github.com/briansmith/ring/pull/425 to see if that PR fixes the NEON detection problem.